beacon_node/beacon_chain/src/block_verification.rs

```diff
@@ -632,6 +637,36 @@ pub fn signature_verify_chain_segment<T: BeaconChainTypes>(
         &chain.spec,
     )?;
 
+    // verify signatures before matching blocks and data
```

Comment:
Feel like we could do the cheaper block & blobs pairing before attempting signature and kzg verification - could we even lift the matching (currently in `da_checker.verify_kzg_for_rpc_blocks`) out of this function?

Reply:
We should not do that. Consider the case:
- block peer sends us bad signature
- column peer sends us valid signature
- because they don't match we penalize column peer

Reply:
ok make sense 👍